bears/general/FilenameBear.py

```diff
@@ -19,14 +20,20 @@ class FilenameBear(LocalBear):
                           'snake': to_snakecase,
                           'space': to_spacecase}
 
+    _language_naming_convention = {'.java': 'pascal',
+                                   '.js': 'kebab',
+                                   '.py': 'snake'}
```

Comment:
https://github.com/AlejandroFrias/case-conversion might have one. If not, they might be a good maintainer of a simple mapping.
coala would eventually have a Language class approach